src/jit/rangecheck.cpp

```diff
@@ -1144,9 +1154,11 @@ Range RangeCheck::ComputeRange(BasicBlock* block, GenTree* expr, bool monotonic
             JITDUMP("%s\n", range.ToString(m_pCompiler->getAllocatorDebugOnly()));
         }
     }
-    else if (varTypeIsSmallInt(expr->TypeGet()))
+    else if (expr->OperIs(GT_CAST) || varTypeIsSmallInt(expr->TypeGet()))
```

Comment:
It looks unnatural for me to have a special case for the cast here, could you please extract the code below into `Range ComputeRageForSmallInt(var_types type)` and call it here and from a new `else if (expr->OperIs(GT_CAST))`?

Note: right now we hit this case only for `IND` and `COMMA` that are based on these indir nodes, e.g.:
```
N002 (  4,  3) [000725] *--X---N----              *  IND       ushort <l:$4c1, c:$4c0>
N001 (  1,  1) [000037] ------------              \--*  LCL_VAR   byref  V23 tmp12        u:1 <l:$140, c:$180>
```

```
N007 (  7,  7) [000248] ---XG--N----              *  COMMA     ubyte  <l:$41b, c:$41a>
N001 (  0,  0) [000505] ------------              +--*  NOP       void  
N006 (  7,  7) [000433] *--XG-------              \--*  IND       ubyte  <l:$419, c:$418>
N005 (  3,  4) [000247] ----G--N----                 \--*  ADD       byref  <l:$1cc, c:$1cb>
N002 (  1,  1) [000246] ------------                    +--*  LCL_VAR   byref  V13 tmp3         u:1 <l:$100, c:$140>
N004 (  2,  3) [000243] ------------                    \--*  CAST      long <- int $448
N003 (  1,  1) [000238] ------------                       \--*  LCL_VAR   int    V07 loc4         u:2 $206
```


@dotnet/jit-contrib does anybody know any other nodes that could have `varTypeIsSmallInt(expr->TypeGet())`?
I expected `GT_CAST` to have them, but as I was told for `GT_CAST`  `TypeGet()` returns `castFrom` type and it can't be less then `INT`, does it sound right?